File: Server/Models/GrowLog.cs
namespace Server.Models
{
    public class GrowLog
    {
        public int Id { get; set; }
        public decimal? AirTemperature { get; set; }
        public decimal FinalPH { get; set; }
        public int FinalPPM { get; set; }
        public decimal? GrowMediumTemperature { get; set; }
        public decimal? Humidity { get; set; }
        public decimal InitialPH { get; set; }
        public int InitialPPM { get; set; }
        public decimal? LightHeight { get; set; }
        public DateTime LogDate { get; set; }
        public string? Notes { get; set; }
        public virtual IEnumerable<AdditiveAdjustment> AdditiveAdjustments { get; set; }
        public int PlantAge { get; set; }
        public decimal? PlantHeight { get; set; }
        public string? Tags { get; set; }
    }
}

// Initial PH, Initial PPM, (Add Nutes, Check PPM), (Check PH, (Optional) PH Adjust)

File: Client/src/models/GrowLog.ts
import { toShortDate } from "../util/functions";
import AdditiveAdjustment from "./AdditiveAdjustment";

export default class GrowLog {
    public id: number = 0;
    public plantId: number = 0;
    public plantAge: number = 0;
    public logDate: string = toShortDate(new Date())!;

    public initialPH: number = 0;
    public initialPPM: number = 0;

    public finalPH: number = 0;
    public finalPPM: number = 0;
    public nutrientAdjustment?: AdditiveAdjustment;
    public phAdjustment?: AdditiveAdjustment;

    public lightHeight?: number;
    public plantHeight?: number;

    public airTemperature?: number;
    public humidity?: number;
    public growMediumTemperature?: number;

    public notes?: string;
    public tags?: string;
}